include/SFML/Graphics/VertexBuffer.hpp

```diff
@@ -0,0 +1,389 @@
+    /// // draw OpenGL stuff that use no vertex buffer...
+    /// \endcode
+    ///
+    /// \param vertexBuffer Pointer to the vertex buffer to bind, can be null to use no vertex buffer
+    ///
+    ////////////////////////////////////////////////////////////
+    static void bind(const VertexBuffer* vertexBuffer);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Set the type of primitives to draw
+    ///
+    /// This function defines how the vertices must be interpreted
+    /// when it's time to draw them:
+    /// \li As points
+    /// \li As lines
+    /// \li As triangles
+    /// \li As quads
+    /// The default primitive type is sf::Points.
+    ///
+    /// \param type Type of primitive
+    ///
+    ////////////////////////////////////////////////////////////
+    void setPrimitiveType(PrimitiveType type);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the type of primitives drawn by the vertex buffer
+    ///
+    /// \return Primitive type
+    ///
+    ////////////////////////////////////////////////////////////
+    PrimitiveType getPrimitiveType() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Set the usage specifier of this vertex buffer
+    ///
+    /// This function provides a hint about how this vertex buffer is
+    /// going to be used in terms of data update frequency.
+    ///
+    /// If data is going to be updated once or more every frame,
+    /// set the usage to Stream. If data is going to be set once
+    /// and used for a long time without being modified, set the
+    /// usage to Static. For everything else Dynamic should be a
+    /// good compromise.
+    ///
+    /// After changing the usage specifier, the vertex buffer has
+    /// to be updated with new data for the usage specifier to
+    /// take effect.
+    ///
+    /// The default primitive type is sf::VertexBuffer::Stream.
+    ///
+    /// \param usage Usage specifier
+    ///
+    ////////////////////////////////////////////////////////////
+    void setUsage(Usage usage);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the usage specifier of this vertex buffer
+    ///
+    /// \return Usage specifier
+    ///
+    ////////////////////////////////////////////////////////////
+    Usage getUsage() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Tell whether or not the system supports vertex buffers
+    ///
+    /// This function should always be called before using
+    /// the vertex buffer features. If it returns false, then
+    /// any attempt to use sf::VertexBuffer will fail.
+    ///
+    /// \return True if vertex buffers are supported, false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    static bool isAvailable();
+
+private:
+
+    friend class RenderTarget;
```

Comment:
Any reason for that? As far as I can tell, `RenderTarget::draw` only uses members that have a corresponding public getter.